crates/bevy_pbr/src/atmosphere/functions.wgsl

```diff
@@ -332,3 +326,64 @@ fn zenith_azimuth_to_ray_dir(zenith: f32, azimuth: f32) -> vec3<f32> {
     let cos_azimuth = cos(azimuth);
     return vec3(sin_azimuth * sin_zenith, mu, -cos_azimuth * sin_zenith);
 }
+
+fn get_density(medium: GpuMedium, altitude: f32) -> f32 {
+    let density_type = u32(medium.density_params.w);
+    switch density_type {
+        case 0u: { // Exponential
+            let scale_height = medium.density_params.x;
+            return exp(-altitude / scale_height);
+        }
+        case 1u: { // SplitExponential
+            let scale_height_lower = medium.density_params.x;
+            let center_altitude = medium.density_params.y;
+            let scale_height_upper = medium.density_params.z;
+            
+            if altitude < center_altitude {
+                return exp(-altitude / scale_height_lower);
+            } else {
+                return exp(-(altitude - center_altitude) / scale_height_upper);
+            }
+        }
+        case 2u: { // Tent
+            let center_altitude = medium.density_params.x;
+            let half_width = medium.density_params.y * 0.5;
+            let exponent = medium.density_params.z;
+            
+            let dist = abs(altitude - center_altitude);
+            return pow(max(0.0, 1.0 - dist / half_width), exponent);
+        }
+        default: {
+            return 0.0;
+        }
+    }
+}
+
+fn get_phase_function(medium: GpuMedium, cos_theta: f32) -> f32 {
+    let phase_type = u32(medium.phase_params.w);
+    switch phase_type {
+        case 0u: { // Rayleigh
+            return rayleigh(cos_theta);
+        }
+        case 1u: { // HenyeyGreenstein
+            let g = medium.phase_params.x;
+            return henyey_greenstein(cos_theta, g);
+        }
+        case 2u: { // CornetteShanks
+            let g = medium.phase_params.x;
+            return cornette_shanks(cos_theta, g);
+        }
+        case 3u: { // DualLobe
+            let g1 = medium.phase_params.x;
+            let g2 = medium.phase_params.y;
+            return mix(
+                henyey_greenstein(cos_theta, g1),
+                henyey_greenstein(cos_theta, g2),
+                0.5
```

Comment:
Should somehow include this in the CPU side parameterization as per eq 18 in 2016 Frostbite Hillarie paper.

$p_{dual}(\theta, g_0, g_1, w) = \text{lerp}(p_{hg}(\theta, g_0), p_{hg}(\theta, g_1), w)$